presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java

```diff
@@ -307,6 +306,22 @@ public boolean supportClientTransaction()
         return roles.build();
     }
 
+    private static Map<String, String> parseExtraCredentials(HttpServletRequest servletRequest)
+    {
+        return parseProperty(servletRequest, PRESTO_EXTRA_CREDENTIAL);
+    }
+
+    private static Map<String, String> parseProperty(HttpServletRequest servletRequest, String headerName)
+    {
+        Map<String, String> properties = new HashMap<>();
+        for (String header : splitSessionHeader(servletRequest.getHeaders(headerName))) {
+            List<String> nameValue = Splitter.on('=').limit(2).trimResults().splitToList(header);
+            assertRequest(nameValue.size() == 2, "Invalid %s header", headerName);
```

Comment:
This check is not needed due to the `limit(2)` above.

Reply:
Instead of using `limit`, I would check if list has length 2, otherwise throw `invalid header` error.

Reply:
We need some clean up in other methods. I see quite a few places where we use `limit(2)` and assert `.size() == 2`

I'll do it in separate PR.